From 09c809c8bcf5c7a96dc719ea6075fec3ee4d6e9d Mon Sep 17 00:00:00 2001
From: wangwudi <wangwudi@hisilicon.com>
Date: Mon, 12 Jun 2023 22:51:24 +0800
Subject: [PATCH 467/960] downstream: irqchip: gic-v3: Collection table support
 muti pages

Only one page is allocated to the collection table.
Recalculate the page number of collection table based on the number of
CPUs.

Signed-off-by: wangwudi <wangwudi@hisilicon.com>
Signed-off-by: Qi Xi <xiqi2@huawei.com>
Signed-off-by: chenyi <chenyi211@h-partners.com>
Signed-off-by: Liu Shixin <liushixin2@huawei.com>
---
 drivers/irqchip/irq-gic-v3-its.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/irqchip/irq-gic-v3-its.c b/drivers/irqchip/irq-gic-v3-its.c
index 77b9b7351f34..0485a0d95262 100644
--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -2642,6 +2642,10 @@ static int its_alloc_tables(struct its_node *its)
 			indirect = its_parse_indirect_baser(its, baser, &order,
 							    ITS_MAX_VPEID_BITS);
 			break;
+		case GITS_BASER_TYPE_COLLECTION:
+			indirect = its_parse_indirect_baser(its, baser, &order,
+							order_base_2(num_possible_cpus()));
+			break;
 		}
 
 		err = its_setup_baser(its, baser, cache, shr, order, indirect);
-- 
2.43.0

